pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BalancedInstanceSelector.java

```diff
@@ -89,7 +91,13 @@ Pair<Map<String, String>, Map<String, String>> select(List<String> segments, int
         if (candidates == null) {
           continue;
         }
-        int selectedIdx = requestId++ % candidates.size();
+        int selectedIdx;
+        if (_useStickyRouting || QueryOptionsUtils.isUseStickyRouting(queryOptions)) {
```

Comment:
We should allow query option to override this. Basically we can have broker config -> table config -> query options. Each level can override previous level if explicitly configured.

Reply:
Added the support for overridng brokerConf with query options. 

The problem I see with table level config is, we either have to fetch the table config on each query, or we need to somehow recreate the routing config each time table config changes or we need to call out that updating the table config setting for this feature needs a broker restart. Thoughts?

Reply:
We use the table config when creating the routing. When table config is changed, it will trigger a routing rebuild to pick up the changes. We rely on the same logic for other routing configs.

Reply:
Makes sense. I did not notice the `RefreshTableConfigMessageHandler`. Changed